### PR TITLE
added new config to limit setting password by presence/value of an at…

### PR DESCRIPTION
**Pull request by fillup**
…tribute on user

### CODE REVIEW

File: src/Ldap.php [hunk 1]
```diff
@@ -75,6 +82,15 @@ class Ldap extends Component implements PasswordStoreInterface
     /** @var \Adldap\Adldap LDAP client*/
     public $ldapClient;
 
+    /** @var \Psr\Log\LoggerInterface */
+    public $logger;
+
+    public function init()
+    {
+        parent::init(); // TODO: Change the autogenerated stub
```
Comment (wcjr) [inline src/Ldap.php]: probably can remove the TODO

File: src/Ldap.php [hunk 2]
```diff
@@ -221,9 +253,39 @@ public function set($employeeId, $password)
...
 
+        $this->logger->notice(sprintf(
+            "Password updated in LDAP for user %s",
+            $logIdentifier
+        ));
+
         return $this->getMeta($employeeId);
     }
 
+    public function matchesRequiredAttributes($user)
```
Comment (Baggerone) [inline src/Ldap.php]: It might be better to have a doc comment.

File: tests/LdapTest.php [hunk 3]
```diff
@@ -116,6 +116,75 @@ public function testAccountDisabled()
         $userMeta = $ldap->getMeta('10121');
     }
```
Comment (Baggerone) [inline tests/LdapTest.php]: Maybe include a test for when there is no updatePasswordIfAttributeAndValue set.

Comment (forevermatt) [reply tests/LdapTest.php]: He did end up adding one and pushing it

**Pull request merged**